rtree.go

```diff
@@ -146,10 +140,11 @@ func (tree *Rtree) bulkLoad(objs []Spatial) {
 	S := math.Floor(math.Sqrt(s))
 
 	// sort all entries by first dimension
-	sortByDim(int(h)%tree.Dim, entries)
-	tree.root = tree.omt(int(h), int(S), entries, int(s))
-	tree.size = n
+	sortByDim(0, entries)
```

Comment:
I'm not sure sorting by x-coordinate first is required. Two points that came to my mind:

1. The paper describes the algorithm for two dimensional data. The R-tree in this library is n-dimensional. 

2. The wording in the paper is `To do this, we need to sort the data items by turns x and y-coordinate in each other level. As we started to sort them by x-coordinate, in level h-2 we sort the data items by y-coordinate and in level h-4, we sort them by x-coordinate and so on.` I understand it as them trying to describe the general approach; to alternate the sorting dimension on every level. Sorting by x-coordinate seems to be an example rather than a requirement.

Reply:
You're right, sorting by the x dimension first is not required. But first, the [original code also sorts by the x dimension first](https://github.com/lovoo/rtreego/blob/master/rtree.go#L162). And second, the sort dimension for a certain level does not differ from tree to tree, depending in the tree height. This makes it more consistent in my eyes.

Furthermore, the SRT paper Diogo mentioned also starts sorting by the first coordinate, then the second, then the third, and so on. So it should work for arbitrary dimensions, too.

Reply:
Yeah, point (2) should be fine.

I am still however concerned regarding @SamiHiltunen's point (1). Btw, Sami implemented the original OMT code in rtreego @tsne.